python_modules/dagster/dagster/core/asset_defs/decorators.py

```diff
@@ -168,6 +169,11 @@ def __call__(self, fn: Callable) -> AssetsDefinition:
             else None,
         )
 
+        # temporary workaround to retrieve asset definition from job
+        op.output_defs[0]._assets_definition = assets_def
```

Comment:
@prha I added the `assets_definition` experimental argument to `Out`/`OutputDefinition`. Since `AssetsDefinition` accepts the `Out` object to build mappings to downstream assets, `Out` has to be created before and passed into `AssetsDefinition`, so I set the instance variable here. Let me know how this looks

Reply:
We should avoid cyclic dependencies and mutable definitions at all costs.

Do we need the full asset definition on the output, or is it sufficient to put the partitions def on it?  If the former, we need to be able to do some lookup by asset key or something to get the definition rather than putting it on the output definition itself.

cc @sryza.  Might be speaking to what is built on top of what.

Reply:
For now, we only need the partitions def, so I can add only the partitions def to the output. However, I suspect that in the future we'll need to access the asset definition from the job